tools/rewriter/GenCallAsm.cpp

```diff
@@ -637,6 +898,45 @@ std::string emit_asm_wrapper(const CAbiSignature &sig,
     |addr |stack by the call to the wrapped function.
     +-----+
 
+    For AArch64:
+    +-----+
+    | top |Top of the stack (stack grows down on AArch64). This address is
+    |     |aligned to 16 bytes.
+    +-----+
+    |ind  |Padding for alignment prior to the indirect args
+    |args |as such memory must be 16B aligned. (TODO: verify this)
```

Comment:
The padding at the start of the stack frame seems unnecessary since this is a public interface, right?
```
The stack must also conform to the following constraint at a public interface:
• SP mod 16 = 0. The stack must be quad-word aligned.
```

Reply:
It's for aligning the indirect args space becuase the stack grows down. The start of each indirect arg region must be aligned to the correct alignment for that arg though, so maybe we don't need a separate slot here to do this.

Reply:
oh I right I forgot there are potentially multiple indirect args. I think if it makes the implementation easier and bugs less likely then a padding slot per arg is preferable for now.